### PR TITLE
I#49 Patch-vendor

### PR DESCRIPTION
**Pull request by ryanknauer**
#49

### CODE REVIEW

File: api-vendor/functions/patch/index.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        let cleanKey = key.split(':')[1];
+        body[cleanKey] = val;
+    });
+
+    Promise.coroutine(function* () {
+        const user_id = event.pathParameters.user_id;
+        let VendorObj = new Vendor().convertToClassObj(body);
+ 
+        let userPrependComma = false;
+        const update_base_user_query = SQL `UPDATE base_user SET `;
```
Comment (anmolg) [inline api-vendor/functions/patch/index.js]: this should be var, since we will be changing it

File: api-vendor/functions/patch/index.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            update_vendor_query.append(SQL `address_y = ${VendorObj._contact_address._location._y}`);
+            vendorPrependComma = true;
+        }
+        update_vendor_query.append(SQL ` WHERE user_id = ${user_id}`);
+
+
+        var client = yield pool.connect();
+        try {
+            yield client.query('BEGIN');
+            let resultUser = yield client.query(update_base_user_query);
```
Comment (anmolg) [inline api-vendor/functions/patch/index.js]: we need to have a check something was even updated. ie, try out this with an empty body and see if the function errors out

File: api-vendor/functions/patch/index.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+            let resultVendor = yield client.query(update_vendor_query);
```
Comment (anmolg) [inline api-vendor/functions/patch/index.js]: we need to have a check something was even updated. ie, try out this with an empty body and see if the function errors out

File: api-vendor/serverless.yml [hunk 4]
```diff
@@ -47,4 +47,15 @@ functions:
...
-                #     arn:
\ No newline at end of file
+                #     arn:
+    # sls invoke local --function PatchVendor --path functions/patch/event.json -s local
+    PatchVendor:
+        handler: functions/patch/index.handler
+        events:
+            - http:
+                path: vendor/{user_id}
+                method: post
```
Comment (anmolg) [inline api-vendor/serverless.yml]: patch

File: api-vendor/functions/patch/index.js (hunk 1 above)
Comment (ryanknauer) [inline api-vendor/functions/patch/index.js]: referring to the prepend comma or the query? Or both? the comma should definitely be. What is the rule if we are appending on to something?

**Pull request merged**